### PR TITLE
Feature/python3.8

### PR DESCRIPTION
**Pull request by mvukov**
I pre-compiled Blender' bpy for Python3.8 and uploaded it on a pre-release https://github.com/oqton/blenderless/releases/tag/bpy-2.92.0-python3.8 (such that we can download it here in this PR). I also updated broken Bazel tooling such that I can run tests. A test is failing, though:

``` bash
bazel test //tests:tests --test_output=streamed 
INFO: Invocation ID: 26248b2e-9817-4716-b97f-b81f33e72260
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Analyzed target //tests:tests (0 packages loaded, 0 targets configured).
INFO: Found 1 test target...
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/mvukov/.cache/bazel/_bazel_mvukov/297d05c0842cc430cba619429642f8f0/sandbox/linux-sandbox/7/execroot/blenderless/bazel-out/k8-fastbuild/bin/tests/tests.runfiles/blenderless
plugins: hydra-core-1.2.0
collected 23 items

tests/test_camera.py .                                                   [  4%]
tests/test_cli.py ...                                                    [ 17%]
tests/test_config.py ..                                                  [ 26%]
tests/test_geometries.py .F...                                           [ 47%]
tests/test_material.py ...                                               [ 60%]
tests/test_render.py ..                                                  [ 69%]
tests/test_scene.py .......                                              [100%]

=================================== FAILURES ===================================
__________________________ test_render_transformation __________________________

    def test_render_transformation():
        verts = np.eye(3, dtype=np.float32)
        faces = np.array([[0, 1, 2]])
        transformation = np.diag([2., 3., 4., 1.])
        transformation[0, 3] = 1.
    
        t_mesh = trimesh.Trimesh(vertices=verts, faces=faces)
    
        # Use MagicMock to avoid loading bpy, makes test slightly overcomplicated
        bpy = MagicMock()
        bpy.data.meshes.new.ret_val = None
        b_mesh = Mesh(mesh=t_mesh, transformation=transformation)
        b_mesh.object_data(bpy)
    
        gt_verts = np.array([[3., 0., 0.], [1., 3., 0.], [1., 0., 4.]])
    
        # Instrument how the vertices are passed to bpy
        called_verts = b_mesh._object_data.from_pydata.call_args[0][0]
    
        # Trimesh stores points in reverse order
>       npt.assert_allclose(gt_verts[::-1, :], called_verts)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 4 / 9 (44.4%)
E       Max absolute difference: 4.
E       Max relative difference: 2.
E        x: array([[1., 0., 4.],
E              [1., 3., 0.],
E              [3., 0., 0.]])
E        y: array([[3., 0., 0.],
E              [1., 3., 0.],
E              [1., 0., 4.]])

tests/test_geometries.py:46: AssertionError
=============================== warnings summary ===============================
tests/test_config.py::test_load_scene_from_config[example_path0]
tests/test_config.py::test_load_scene_from_config[example_path1]
  /home/mvukov/.cache/bazel/_bazel_mvukov/297d05c0842cc430cba619429642f8f0/sandbox/linux-sandbox/7/execroot/blenderless/bazel-out/k8-fastbuild/bin/tests/tests.runfiles/blenderless/tests/test_config.py:33: FutureWarning: `multichannel` is a deprecated argument name for `structural_similarity`. It will be removed in version 1.0. Please use `channel_axis` instead.
    assert structural_similarity(base_render, test_render, multichannel=True) > 0.995

tests/test_render.py::test_gif
tests/test_render.py::test_gif
tests/test_render.py::test_gif
tests/test_render.py::test_gif
tests/test_render.py::test_gif
tests/test_scene.py::test_render_gif
tests/test_scene.py::test_render_gif
  /home/mvukov/.cache/bazel/_bazel_mvukov/297d05c0842cc430cba619429642f8f0/sandbox/linux-sandbox/7/execroot/blenderless/bazel-out/k8-fastbuild/bin/tests/tests.runfiles/blenderless/blenderless/scene.py:77: DeprecationWarning: Starting with ImageIO v3 the behavior of this function will switch to that of iio.v3.imread. To keep the current behavior (and make this warning dissapear) use `import imageio.v2 as imageio` or call `imageio.v2.imread` directly.
    images.append(imageio.imread(filename))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_geometries.py::test_render_transformation - AssertionError: 
================== 1 failed, 22 passed, 9 warnings in 49.94s ===================
FAIL: //tests:tests (see /home/mvukov/.cache/bazel/_bazel_mvukov/297d05c0842cc430cba619429642f8f0/execroot/blenderless/bazel-out/k8-fastbuild/testlogs/tests/tests/test.log)
Target //tests:tests up-to-date:
  bazel-bin/tests/tests_runner.py
  bazel-bin/tests/tests
INFO: Elapsed time: 52.841s, Critical Path: 52.21s
INFO: 2 processes: 2 linux-sandbox.
INFO: Build completed, 1 test FAILED, 2 total actions
//tests:tests                                                            FAILED in 51.3s
  /home/mvukov/.cache/bazel/_bazel_mvukov/297d05c0842cc430cba619429642f8f0/execroot/blenderless/bazel-out/k8-fastbuild/testlogs/tests/tests/test.log

INFO: Build completed, 1 test FAILED, 2 total actions
```
Do the tests take too long?

@vlaminckaxel @rubenverhack Can you guys take a look at the failing test?

### CODE REVIEW

File: blenderless/scene.py [hunk 1]
```diff
@@ -19,8 +19,8 @@ def import_bpy():
     The reason for this late-loading is that communication with Blender is
     restricted to a single separate render thread.
     """
-    os.environ["BLENDER_SYSTEM_SCRIPTS"] = 'external/blenderpy/bpy-2.91a0.data/scripts/2.91/scripts'
-    os.environ["BLENDER_SYSTEM_DATAFILES"] = 'external/blenderpy/bpy-2.91a0.data/scripts/2.91/datafiles'
+    os.environ["BLENDER_SYSTEM_SCRIPTS"] = 'external/bpy/2.92/scripts'
+    os.environ["BLENDER_SYSTEM_DATAFILES"] = 'external/bpy/2.92/datafiles'
```
Comment (mvukov) [inline blenderless/scene.py:23]: NOTE: looks like this is not needed any more.

Comment (rubenverhack) [reply blenderless/scene.py:23]: That would be great since this was quite a dirty fix.

**Comment by rubenverhack** [comment]
```
        # Trimesh stores points in reverse order
>       npt.assert_allclose(gt_verts[::-1, :], called_verts)
``` 

It seems like the current behavior is actually makes more sense the expected behavior. The points don't seem to be stored in reverse order any more so the fix simply would be 
```
      npt.assert_allclose(gt_verts, called_verts)
``` 

The question is: what caused this change? I believe you only changed bpy and bpy is explicitely not being loaded :) Was there a bump in trimesh versions? 

I do remember writing this test and not understanding why trimesh was storying points in reverse order but it just seemed to be that way.

**Comment by rubenverhack** [comment]
The assumption could be tested by adding a numpy array of vertices to trimesh and then retrieve that attribute back from trimesh to see if it's reversed or not. @mvukov I can't seem to access the DGX from where I am at the moment (see DGX slack channel).

File: blenderless/scene.py (hunk 1 above)
Comment (mvukov) [reply blenderless/scene.py]: OK, I removed this.

**Comment by mvukov** [comment]
> The question is: what caused this change? I believe you only changed bpy and bpy is explicitely not being loaded :) Was there a bump in trimesh versions?

I think it may be trimesh-version related. I set  the same constraint now on trimesh version (in requirements.txt) as we use internally.

**Comment by rubenverhack** [comment]
And did it change the test? 

TBH I don't think it matters (the order), but it would be good to get @vlaminckaxel final blessing on this since he's deepest in this. I only rewrote this test but don't really know if this functionality is used and/or the order of the points actually matter. He's back from holidays on Monday.

**Comment by mvukov** [comment]
In master we use trimesh "3.9.29" and here we use "3.14.0".

**Comment by mvukov** [comment]
> And did it change the test?

IMO, this version change is relevant.

**Comment by mvukov** [comment]
@vlaminckaxel Keep in mind that poetry-based build still works with Python3.7 and old blenderpy.

**Review by vlaminckaxel — APPROVED** [review]
lgtm!

**Pull request merged**